GitHubRepositories.ps1

```diff
@@ -760,8 +816,12 @@ function Update-GitHubRepository
     .EXAMPLE
         Update-GitHubRepository -OwnerName Microsoft -RepositoryName PowerShellForGitHub -Description 'The best way to automate your GitHub interactions'
 
+        Changes the description of the specified repository.
+
     .EXAMPLE
-        Update-GitHubRepository -Uri https://github.com/PowerShell/PowerShellForGitHub -Private:$false
+        Update-GitHubRepository -Uri https://github.com/PowerShell/PowerShellForGitHub -Visibility 'Private'
+
+        Changes the visibility of the specified repository to 'Private'.
```

Comment:
I think this one is a bad example that won't work.  The example that you're replacing was changing a repo to to be _Public_ (`-Private:$false`).  I don't think that GitHub _lets_ you change a repo to be Private once it's already public.

```suggestion
        Update-GitHubRepository -Uri https://github.com/PowerShell/PowerShellForGitHub -Visibility 'Public'

        Changes the visibility of the specified repository to be public.
```

Reply:
You can do it from the GUI, though I've just tried it from the function, and it didn't error, but hasn't changed the repo's state.

By the way, I am receiving the following error when calling any functions from the current master branch:

```
Microsoft.PowerShell.Commands.HttpResponseException: Response status code does not indicate success: 400 (Invalid instrumentation key).
   at System.Management.Automation.MshCommandRuntime.ThrowTerminatingError(ErrorRecord errorRecord)
```

Reply:
Interesting. I could have sworn that once it was public, you couldn't make it private again...maybe that's only if there are forks.  I still think it would be better to keep the example going to public than to private.

Reply:
In the GUI it comes up with a big warning with a list of features that will be lost going from Public to Private. I changed the example anyway.

Reply:
As for the error, I can't reproduce it.  It _sounds_ like it's related to #186, but it's not repro'ing for me.  My best guess is that you've modified the `ApplicationInsightsKey` configuration property, and so you're seeing that error when it's trying to report the telemetry.

To verify:
```powershell
Get-GitHubConfiguration -Name ApplicationInsightsKey
```

The expected value should be:
https://github.com/microsoft/PowerShellForGitHub/blob/d1bd976d70cc975dfd247f9ad2bace58a465c7da/GitHubConfiguration.ps1#L611

If you're not seeing that, then you have a bunch of options ahead of you to fix it:
1. `Set-GitHubConfiguration -ApplicationInsightsKey '<that value>'`  (this will make it work as expected)
2. `Reset-GitHubConfiguration` (this will lose all your current settings and get all the default ones)
3. `Set-GitHubConfiguration -DisableTelemetry` (this will ignore that invalid ApplicationInsightsKey because it will just stop reporting telemetry)

Reply:
OK `Reset-GitHubConfiguration` fixed it, and I know how I broke it. I ran `Set-GitHubConfiguration` and missed out the 'dash' prefix on the parameter name I wanted to set, which because none of the functions have `PositionalBinding=$false` set, overwrote the first two properties (`ApiHostName` and `ApplicationInsightKey`).
My opinion is functions should always have `PositionalBinding=$false` set and if required, the `position` property set on individual parameters.

Reply:
Didn't even know `PositionalBinding=$false` was a thing.  You're not the first one to get bitten on a similar type of issue for `Set-GitHubConfiguration`.  Of all the functions, that one is ripe for that value.  My only concern is that, [based on the docs](https://docs.microsoft.com/en-us/dotnet/api/system.management.automation.cmdletbindingattribute.positionalbinding), that's not available in PS 4.0 which we still need to support.

Reply:
I thought `PositionalBinding` was implemented with Advanced Functions back in PowerShell 3, but I couldn't say for sure. Why does the module still need to support PowerShell 4?

Reply:
Some internal systems that this module is used on are still running on older versions of Windows Server which are still using PS4, and it's not within our power to update those servers nor to install PS7Core.  At some point those servers will get updated, but until then:

https://github.com/microsoft/PowerShellForGitHub/blob/d1bd976d70cc975dfd247f9ad2bace58a465c7da/PowerShellForGitHub.psd1#L48-L49

Reply:
I've opened #219 to track this.